R/matrix_form.R

```diff
@@ -15,24 +15,36 @@ mform_handle_newlines <- function(matform) {
   has_topleft <- mf_has_topleft(matform)
   strmat <- mf_strings(matform)
   frmmat <- mf_formats(matform)
+  spamat <- mf_spans(matform)
+  alimat <- mf_aligns(matform)
   nr_header <- mf_nrheader(matform)
+  nl_inds_header <- seq(1, mf_nlheader(matform))
   hdr_inds <- 1:nr_header
 
   # hack that is necessary only if bottom aligned
   topleft_has_nl_char <- FALSE
   if (has_topleft) {
-    tl <- strmat[hdr_inds, 1, drop = TRUE]
-    strmat[hdr_inds, 1] <- ""
+    tl <- strmat[nl_inds_header, 1, drop = TRUE]
+    strmat[nl_inds_header, 1] <- ""
     tl <- tl[nzchar(tl)] # we are not interested in initial "" but we cover initial \n
     topleft_has_nl_char <- any(grepl("\n", tl))
     tl_to_add_back <- strsplit(paste0(tl, collapse = "\n"), split = "\n", fixed = TRUE)[[1]]
     how_many_nl <- length(tl_to_add_back)
   }
+
+  # pre-proc in case of wrapping and \n
+  line_grouping <- mf_lgrouping(matform)
+  strmat <- .compress_mat(strmat, line_grouping, "nl")
+  frmmat <- .compress_mat(frmmat, line_grouping, "unique") # never not unique
+  spamat <- .compress_mat(spamat, line_grouping, "unique")
+  alimat <- .compress_mat(alimat, line_grouping, "unique")
+  line_grouping <- unique(line_grouping)
```

Comment:
this compression is needed when columns are wrapped. Without it we get a weird bottom alignment if there are \n preceding this